lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -144,12 +157,9 @@ public function bindParam($column, &$variable, $type = null, $length = null)
     {
         $column = isset($this->_paramMap[$column]) ? $this->_paramMap[$column] : $column;
 
-        if ($type == \PDO::PARAM_LOB) {
-            $lob = oci_new_descriptor($this->_dbh, OCI_D_LOB);
-            $lob->writeTemporary($variable, OCI_TEMP_BLOB);
-
-            return oci_bind_by_name($this->_sth, $column, $lob, -1, OCI_B_BLOB);
-        } elseif ($length !== null) {
+        if (is_a($variable, 'Oci-Lob')) {
```

Comment:
`Oci-Lob` can't stay as magic constant. Also, seems weird that the OCI extension would provide a type with a dash in it.